contracts/bridge/Bridge.sol

```diff
@@ -84,47 +84,93 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
     );
 
     // start allows the value transfer request.
-    function start() external onlyOwner {
-        isRunning = true;
-    }
-
-    // stop prevent the value transfer request.
-    function stop() external onlyOwner {
-        isRunning = false;
+    function start(bool _status)
+        external
+        onlyOwner
+    {
+        isRunning = _status;
     }
 
     // stop prevent the value transfer request.
-    function setCounterPartBridge(address _bridge) external onlyOwner {
+    function setCounterPartBridge(address _bridge)
+        external
+        onlyOwner
+    {
         counterpartBridge = _bridge;
     }
 
     // registerToken can update the allowed token with the counterpart token.
-    function registerToken(address _token, address _cToken) external onlyOwner {
+    function registerToken(address _token, address _cToken)
+        external
+        onlyOwner
+    {
         allowedTokens[_token] = _cToken;
     }
 
     // deregisterToken can remove the token in allowedToken list.
-    function deregisterToken(address _token) external onlyOwner {
+    function deregisterToken(address _token)
+        external
+        onlyOwner
+    {
         delete allowedTokens[_token];
     }
 
+    // registerOperator registers a new operator.
+    function registerOperator(address _operator)
+        external
+        onlyOwner
+    {
+        operators[_operator] = true;
+    }
+
+    // deregisterOperator deregisters the operator.
+    function deregisterOperator(address _operator)
+        external
+        onlyOwner
+    {
+        delete operators[_operator];
+    }
+
+    // setOperatorThreshold sets the operator threshold.
+    function setOperatorThreshold(VoteType _voteType, uint64 _threshold)
+        external
+        onlyOwner
+    {
+        operatorThresholds[uint8(_voteType)] = _threshold;
+    }
+
+    function updateHandleNonce(uint64 _requestedNonce) internal {
+        uint64 i;
+        handledNonces[_requestedNonce] = true;
+
+        if (_requestedNonce > maxHandledRequestedNonce) {
+            maxHandledRequestedNonce = _requestedNonce;
+        }
+        for (i = sequentialHandleNonce; i <= maxHandledRequestedNonce && handledNonces[i]; i++) { }
+        sequentialHandleNonce = i;
+    }
+
     // handleERC20Transfer sends the token by the request.
     function handleERC20Transfer(
         address _from,
         address _to,
         address _tokenAddress,
         uint256 _value,
-        uint64 _requestNonce,
-        uint64 _requestBlockNumber
+        uint64 _requestedNonce,
+        uint64 _requestedBlockNumber
     )
         public
-        onlyOwner
+        onlyOperators
     {
-        require(handleNonce == _requestNonce, "mismatched handle / request nonce");
+        bytes32 voteKey = keccak256(abi.encodePacked(VoteType.ValueTransfer, _from, _to, _tokenAddress, _value, _requestedNonce, _requestedBlockNumber));
```

Comment:
`voteKey` is always used paring with `_requestedNonce`. 
Is `_requestedNonce` required as an element of `voteKey`?